scaling_performance/network_optimization.adoc

```diff
@@ -67,6 +70,57 @@ xref:../architecture/core_concepts/routes.adoc#router-sharding[Router sharding]
 should be used to serve more routes towards applications and help horizontally
 scale the routing tier.
 
+[[scaling-performance-optimizing-router-haproxy]]
+=== Performance Optimizations
+
+[[scaling-performance-optimizing-router-haproxy-maxconn]]
+==== Setting the Maximum Number of Connections
+
+One of the most important tunable parameters for HAProxy is the 
+`maxconn` parameter which sets the maximum per-process number 
+of concurrent connections to a given number.  In {product-title} 
+HAProxy router this parameter is adjustable by editing 
+the 
+`xref:../install_config/router/default_haproxy_router.adoc#concurrent-connections[ROUTER_MAX_CONNECTIONS]`
+environment variable in the router's
+deployment configuration.
+
+
+[[scaling-performance-optimizing-router-haproxy-cpu-affinity]]
+==== CPU and Interrupt Affinity
+
+While it is entirely possible to run HAProxy with multiple worker
+processes using the `nbproc` parameter, in {product-title} HAProxy
+router runs as a single process.  {product-title} HAProxy router
+will typically perform better on a system with a fewer but high
+frequency cores, rather than on an SMP system with a high number of
+lower frequency cores.
+
+Pinning HAProxy process to one CPU core and the network interrupts to another,
+while all sharing the same L3 cache, tends to increase network performance
+because in practice the amount of work for HAProxy and the network stack are
```

Comment:
Instead of quite close you could say that having processes and interrupts on the same NUMA node helps avoid memory accesses by ensuring a shared L3 cache.  You should also note that this level of control is generally not possible on a public cloud environment.  I'd also mention that irqbalance when running on bare metal automatically handles PCI locality and NUMA affinity for IRQs.  Again on a cloud environment this level of information is generally not provided to the OS.